lib/ansible/module_utils/vmware.py

```diff
@@ -828,41 +831,181 @@ def is_vcenter(self):
+                                                  "or '/'" % self.params['name'])
+                    elif user_folder == '/vm' or user_folder.startswith('/vm/'):
+                        # User provided VMware default vm folder with / or
+                        # User provided nested folder under VMware default vm folder i.e. folder = /vm/india/finance
+                        user_desired_path = "%s%s%s" % (dcpath, user_defined_dc, user_folder)
+                    elif user_folder == 'vm':
+                        # User provided VMware default vm folder
+                        user_desired_path = "%s%s/vm" % (dcpath, user_defined_dc)
+                    elif user_folder.startswith("%s%s/vm" % (dcpath, user_defined_dc)) or \
+                            user_folder.startswith("/%s" % user_defined_dc):
+                        # User defined datacenter is not nested i.e. dcpath = '/' , or
+                        # User defined datacenter is nested i.e. dcpath = '/F0/DC0' or
+                        # User provided folder starts with / and datacenter i.e. folder = /ha-datacenter/
+                        user_desired_path = user_folder
+                    elif user_folder.startswith('/'):
+                        # User defined folder is under VMware default vm folder i.e. folder = /india/finance
+                        user_desired_path = "%s%s/vm%s" % (dcpath, user_defined_dc, user_folder)
+                    elif user_folder.startswith('%s' % user_defined_dc):
+                        # User defined folder is datacenter i.e. folder = DC0
+                        user_desired_path = "%s%s/vm" % (dcpath, user_defined_dc)
+                    else:
+                        # get_vm is not super cow !!!
+                        user_desired_path = user_folder
+
+                    # Check if user has provided same path as virtual machine
+                    actual_vm_folder_path = self.get_vm_path(content=self.content, vm_name=vm)
+                    if user_desired_path in actual_vm_folder_path:
+                        vm_obj = vm
+                        break
+            elif vms:
+                # Unique virtual machine found.
+                vm_obj = vms[0]
+
+        if vm_obj:
+            self.current_vm_obj = vm_obj
+
+        return vm_obj
 
     def gather_facts(self, vm):
+        """
+        Function to gather facts of virtual machine.
+        Args:
+            vm: Name of virtual machine.
+
+        Returns: Facts dictionary of the given virtual machine.
+
+        """
         return gather_vm_facts(self.content, vm)
 
     @staticmethod
-    def get_vm_path(content, vm):
-        foldername = None
-        folder = vm.parent
+    def get_vm_path(content, vm_name):
+        """
+        Function to find the path of virtual machine.
+        Args:
+            content: VMware content object
+            vm_name: Name of the virtual machine
```

Comment:
You should explicit the fact that vm_name is not a string, but a vmodl.… object containing a propSet containing the VM name.

Reply:
OK.